app/controllers/pipeline_viz_controller.rb

```diff
@@ -0,0 +1,40 @@
+class PipelineVizController < ApplicationController
+  before_action :authenticate_user!
+
+  current_power do
+    Power.new(current_user)
+  end
+
+  # GET /sample/:sample_id/pipeline_viz
+  # GET /sample/:sample_id/pipeline_viz.json
+  def show
```

Comment:
Just a note: as a show action in a nested controller this would be: `/sample/:sample_id/pipeline_viz/:pipeline_id`. You can use the pipeline version as the id, and thus you would have support for different pipeline viz within same sample.

Reply:
I added selecting pipeline version into the method!